### PR TITLE
Improvement/fix data saving

### PR DESCRIPTION
**Pull request by chf3kl9**
fixes issues with data saving
previously, if a team joined a tournament, the tournament would know, but the team wouldn't

### CODE REVIEW

**Review by larsvandriel — DISMISSED** [review]

File: src/main/java/com/fontys/api/service/MatchService.java [hunk 1]
```diff
@@ -52,7 +52,10 @@ public Match createMatch(Integer teamHomeId, Integer teamAwayId, String dateStri
 
         Match match = new Match(teamHome, teamAway, date, tournament);
 
-        return matchRepository.save(match);
+        Match match1 = matchRepository.save(match);
```
Comment (lfb0801) [inline src/main/java/com/fontys/api/service/MatchService.java]: naming of variables is weird

File: src/main/java/com/fontys/api/service/TeamService.java [hunk 2]
```diff
@@ -118,7 +120,9 @@ else if (team.isEmpty())
                 return "User is not added to the team!";
             }
             team1.getUsers().remove(user1);
+            user1.getTeams().remove(team1);
```
Comment (lfb0801) [inline src/main/java/com/fontys/api/service/TeamService.java:123]: brown ditto 💩

File: src/main/java/com/fontys/api/service/TeamService.java [hunk 3]
```diff
@@ -118,7 +120,9 @@ else if (team.isEmpty())
...
             teamRepository.save(team1);
+            userRepository.save(user1);
```
Comment (lfb0801) [inline src/main/java/com/fontys/api/service/TeamService.java:125]: brown ditto 💩

File: src/main/java/com/fontys/api/service/TeamService.java [hunk 4]
```diff
@@ -118,7 +120,9 @@ else if (team.isEmpty())
                 return "User is not added to the team!";
             }
             team1.getUsers().remove(user1);
+            user1.getTeams().remove(team1);
             teamRepository.save(team1);
```
Comment (lfb0801) [inline src/main/java/com/fontys/api/service/TeamService.java:124]: brown ditto 💩

File: src/main/java/com/fontys/api/service/TournamentService.java [hunk 5]
```diff
@@ -137,7 +138,9 @@ else if (team.isEmpty())
             return "The tournament is currently filled with teams!";
         }
         tournament1.getTeams().add(team1);
+        team1.getTournaments().add(tournament1);
```
Comment (lfb0801) [inline src/main/java/com/fontys/api/service/TournamentService.java:141]: brown ditto 💩

File: src/main/java/com/fontys/api/service/TournamentService.java [hunk 6]
```diff
@@ -137,7 +138,9 @@ else if (team.isEmpty())
...
         tournamentRepository.save(tournament1);
+        teamRepository.save(team1);
```
Comment (lfb0801) [inline src/main/java/com/fontys/api/service/TournamentService.java:143]: brown ditto 💩

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -1,3 +1,3 @@
-server.port=8080
+server.port=8090
```
Comment (lfb0801) [inline src/main/resources/application.properties]: This choice will brake the frontend, so NAH NOT GOING TO HAPPEN CHIEF

**Review by lfb0801 — CHANGES_REQUESTED** [review]

**Review by lfb0801 — APPROVED** [review]
thanks for fixing the changes

**Review by Zeehond99 — APPROVED** [review]
Looks fine

**Pull request merged**